plugins/GSdx/GSRenderer.cpp

```diff
@@ -463,28 +463,21 @@ void GSRenderer::VSync(int field)
 
 	if(!m_snapshot.empty())
 	{
-		bool shift = false;
-
-		#ifdef _WIN32
-
-		shift = !!(::GetAsyncKeyState(VK_SHIFT) & 0x8000);
-
-		#else
-
-		shift = m_shift_key;
-
-		#endif
-
-		if(!m_dump && shift)
+		if(!m_dump && m_shift_key)
 		{
-			GSFreezeData fd;
-			fd.size = 0;
-			fd.data = NULL;
+			GSFreezeData fd = {0, nullptr};
 			Freeze(&fd, true);
 			fd.data = new uint8[fd.size];
 			Freeze(&fd, false);
 
-			m_dump.Open(m_snapshot, m_crc, fd, m_regs);
+#ifdef LZMA_SUPPORTED
+			if (m_control_key)
+				m_dump = std::unique_ptr<GSDumpBase>(new GSDump(m_snapshot, m_crc, fd, m_regs));
+			else
+				m_dump = std::unique_ptr<GSDumpBase>(new GSDumpXz(m_snapshot, m_crc, fd, m_regs));
+#else
+			m_dump = std::unique_ptr<GSDumpBase>(new GSDump(m_snapshot, m_crc, fd, m_regs));
+#endif
```

Comment:
First Vsync is the next frame, so yes could be verbose. We need to do an initial Vsync when replaying.

I created a Write function to check size and m_gs properly. It should be better now. Is it ok for you ?